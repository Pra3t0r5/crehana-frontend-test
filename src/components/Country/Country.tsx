import React, { FC } from "react";
import {
  Avatar,
  Card,
  CardContent,
  CardHeader,
  Collapse,
  List,
  ListItem,
  ListItemIcon,
  ListItemText,
  ListSubheader,
  Typography,
} from "@mui/material";
import ShortcutIcon from "@mui/icons-material/Shortcut";
import DriveFileRenameOutlineIcon from "@mui/icons-material/DriveFileRenameOutline";
import CurrencyExchangeIcon from "@mui/icons-material/CurrencyExchange";
import PublicIcon from "@mui/icons-material/Public";
import LanguageIcon from "@mui/icons-material/Language";
import LocationCityIcon from "@mui/icons-material/LocationCity";
import ArrowRightIcon from "@mui/icons-material/ArrowRight";
import { Country as CountryType } from "../../generated/graphql";
import "./Country.css";
import { Link } from "react-router-dom";

interface CountryProps {
  country: CountryType;
}

export interface CountryRowProps {
  code: string;
  name: string;
  continent: {
    code: string;
  };
}

const Country: FC<CountryProps> = ({ country }) => {
  return (
    <>
      {/* {isLoading && <p>Loading ...</p>}
      {error && <p>{JSON.stringify(error)}</p>} */}
      {/* <Link to="/" /> */}
      <Card
        sx={{ width: 500, m: "auto" }}
        key={country.code}
        id="country-detail"
        data-testid="Country"
      >
        <CardHeader
          id="country-detail--header"
          avatar={
            <Avatar
              sx={{ bgcolor: "lightgray", height: 80, width: 80, fontSize: 45 }}
              aria-label="flag"
            >
              {country.emoji}
            </Avatar>
          }
          title={country.name}
          titleTypographyProps={{ variant: "h3" }}
          subheader={country.code}
        />
        <CardContent id="country-detail--content">
          <List
            id="country-detail--content--list"
            subheader={
              <ListSubheader component="div">
                Detailed Information
              </ListSubheader>
            }
          >
            <ListItem>
              <ListItemIcon>
                <ShortcutIcon color="primary" />
              </ListItemIcon>
              <ListItemText primary={country.code} />
            </ListItem>
            {/* <ListItem>
              <ListItemIcon>
                <DriveFileRenameOutlineIcon color="primary" />
              </ListItemIcon>
              <ListItemText primary={country.name} />
            </ListItem> */}
            <ListItem>
              <ListItemIcon>
                <CurrencyExchangeIcon color="primary" />
              </ListItemIcon>
              <ListItemText primary={country.currency} />
            </ListItem>
            <ListItem>
              <ListItemIcon>
                <PublicIcon color="primary" />
              </ListItemIcon>
              <ListItemText primary={country.continent?.name} />
            </ListItem>
            <ListItem>
              <ListItemIcon>
                <LanguageIcon color="primary" />
              </ListItemIcon>
              <ListItemText primary="Languages" />
            </ListItem>
            <Collapse in={true} timeout="auto">
              <List
                component="div"
                disablePadding
                id="country-detail--content--list-lang"
              >
                {country.languages?.map((lang) => (
                  <ListItem sx={{ pl: 4 }} key={lang.code}>
                    <ListItemIcon>
                      <ArrowRightIcon />
                    </ListItemIcon>
                    <ListItemText primary={lang.name} />
                  </ListItem>
                ))}
              </List>
            </Collapse>
            <ListItem>
              <ListItemIcon>
                <LocationCityIcon color="primary" />
              </ListItemIcon>
              <ListItemText primary={country.capital} />
            </ListItem>
          </List>
        </CardContent>
      </Card>
    </>
  );
};

export default Country;
