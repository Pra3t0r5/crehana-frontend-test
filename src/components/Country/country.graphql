query Country($code: ID!) {
  country(code: $code) {
    code
    name
    native
    phone
    continent {
      code
      name
    }
    capital
    currency
    languages {
      code
      name
      rtl
    }
    emoji
    emojiU
  }
}
