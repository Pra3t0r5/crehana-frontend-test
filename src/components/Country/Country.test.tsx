import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Country from './Country';
import renderer from 'react-test-renderer';

const countryA = {
  code: 'US',
  name: 'United States',
  native: 'United States',
  phone: '1',
  continent: {
    code: 'NA',
    name: 'North America',
  },
  currency: 'USD',
  languages: [
    {
      code: 'en',
      name: 'English',
      native: 'English',
      rtl: false,
    },
  ],
  emoji: 'ðŸ‡ºðŸ‡¸',
  emojiU: 'U+1F1FA U+1F1F8',
}

describe('<Country />', () => {
  test('it should mount', () => {
    render(<Country country={countryA} />);

    const country = screen.getByTestId('Country');

    expect(country).toBeInTheDocument();
  });
  test('it should match snapshot', () => {
    const tree = renderer.create(<Country country={countryA} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});