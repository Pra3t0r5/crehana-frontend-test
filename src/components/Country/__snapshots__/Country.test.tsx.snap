// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Country /> it should match snapshot 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-71zi9g-MuiPaper-root-MuiCard-root"
  data-testid="Country"
  id="country-detail"
>
  <div
    className="MuiCardHeader-root css-185gdzj-MuiCardHeader-root"
    id="country-detail--header"
  >
    <div
      className="MuiCardHeader-avatar css-1ssile9-MuiCardHeader-avatar"
    >
      <div
        aria-label="flag"
        className="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-1flf7am-MuiAvatar-root"
      >
        ðŸ‡ºðŸ‡¸
      </div>
    </div>
    <div
      className="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
    >
      <span
        className="MuiTypography-root MuiTypography-h3 MuiCardHeader-title css-1thj51z-MuiTypography-root"
      >
        United States
      </span>
      <span
        className="MuiTypography-root MuiTypography-body2 MuiCardHeader-subheader css-83ijpv-MuiTypography-root"
      >
        US
      </span>
    </div>
  </div>
  <div
    className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
    id="country-detail--content"
  >
    <ul
      className="MuiList-root MuiList-padding MuiList-subheader css-p64g7a-MuiList-root"
      id="country-detail--content--list"
    >
      <div
        className="MuiListSubheader-root MuiListSubheader-gutters MuiListSubheader-sticky css-uob957-MuiListSubheader-root"
      >
        Detailed Information
      </div>
      <li
        className="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
        disabled={false}
      >
        <div
          className="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium css-ci5apu-MuiSvgIcon-root"
            data-testid="ShortcutIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="m21 11-6-6v5H8c-2.76 0-5 2.24-5 5v4h2v-4c0-1.65 1.35-3 3-3h7v5l6-6z"
            />
          </svg>
        </div>
        <div
          className="MuiListItemText-root css-tlelie-MuiListItemText-root"
        >
          <span
            className="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
          >
            US
          </span>
        </div>
      </li>
      <li
        className="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
        disabled={false}
      >
        <div
          className="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium css-ci5apu-MuiSvgIcon-root"
            data-testid="CurrencyExchangeIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12.89 11.1c-1.78-.59-2.64-.96-2.64-1.9 0-1.02 1.11-1.39 1.81-1.39 1.31 0 1.79.99 1.9 1.34l1.58-.67c-.15-.45-.82-1.92-2.54-2.24V5h-2v1.26c-2.48.56-2.49 2.86-2.49 2.96 0 2.27 2.25 2.91 3.35 3.31 1.58.56 2.28 1.07 2.28 2.03 0 1.13-1.05 1.61-1.98 1.61-1.82 0-2.34-1.87-2.4-2.09l-1.66.67c.63 2.19 2.28 2.78 2.9 2.96V19h2v-1.24c.4-.09 2.9-.59 2.9-3.22 0-1.39-.61-2.61-3.01-3.44zM3 21H1v-6h6v2H4.52c1.61 2.41 4.36 4 7.48 4 4.97 0 9-4.03 9-9h2c0 6.08-4.92 11-11 11-3.72 0-7.01-1.85-9-4.67V21zm-2-9C1 5.92 5.92 1 12 1c3.72 0 7.01 1.85 9 4.67V3h2v6h-6V7h2.48C17.87 4.59 15.12 3 12 3c-4.97 0-9 4.03-9 9H1z"
            />
          </svg>
        </div>
        <div
          className="MuiListItemText-root css-tlelie-MuiListItemText-root"
        >
          <span
            className="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
          >
            USD
          </span>
        </div>
      </li>
      <li
        className="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
        disabled={false}
      >
        <div
          className="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium css-ci5apu-MuiSvgIcon-root"
            data-testid="PublicIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-1 17.93c-3.95-.49-7-3.85-7-7.93 0-.62.08-1.21.21-1.79L9 15v1c0 1.1.9 2 2 2v1.93zm6.9-2.54c-.26-.81-1-1.39-1.9-1.39h-1v-3c0-.55-.45-1-1-1H8v-2h2c.55 0 1-.45 1-1V7h2c1.1 0 2-.9 2-2v-.41c2.93 1.19 5 4.06 5 7.41 0 2.08-.8 3.97-2.1 5.39z"
            />
          </svg>
        </div>
        <div
          className="MuiListItemText-root css-tlelie-MuiListItemText-root"
        >
          <span
            className="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
          >
            North America
          </span>
        </div>
      </li>
      <li
        className="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
        disabled={false}
      >
        <div
          className="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium css-ci5apu-MuiSvgIcon-root"
            data-testid="LanguageIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zm6.93 6h-2.95c-.32-1.25-.78-2.45-1.38-3.56 1.84.63 3.37 1.91 4.33 3.56zM12 4.04c.83 1.2 1.48 2.53 1.91 3.96h-3.82c.43-1.43 1.08-2.76 1.91-3.96zM4.26 14C4.1 13.36 4 12.69 4 12s.1-1.36.26-2h3.38c-.08.66-.14 1.32-.14 2 0 .68.06 1.34.14 2H4.26zm.82 2h2.95c.32 1.25.78 2.45 1.38 3.56-1.84-.63-3.37-1.9-4.33-3.56zm2.95-8H5.08c.96-1.66 2.49-2.93 4.33-3.56C8.81 5.55 8.35 6.75 8.03 8zM12 19.96c-.83-1.2-1.48-2.53-1.91-3.96h3.82c-.43 1.43-1.08 2.76-1.91 3.96zM14.34 14H9.66c-.09-.66-.16-1.32-.16-2 0-.68.07-1.35.16-2h4.68c.09.65.16 1.32.16 2 0 .68-.07 1.34-.16 2zm.25 5.56c.6-1.11 1.06-2.31 1.38-3.56h2.95c-.96 1.65-2.49 2.93-4.33 3.56zM16.36 14c.08-.66.14-1.32.14-2 0-.68-.06-1.34-.14-2h3.38c.16.64.26 1.31.26 2s-.1 1.36-.26 2h-3.38z"
            />
          </svg>
        </div>
        <div
          className="MuiListItemText-root css-tlelie-MuiListItemText-root"
        >
          <span
            className="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
          >
            Languages
          </span>
        </div>
      </li>
      <div
        className="MuiCollapse-root MuiCollapse-vertical MuiCollapse-entered css-pwcg7p-MuiCollapse-root"
        style={
          Object {
            "minHeight": "0px",
          }
        }
      >
        <div
          className="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
        >
          <div
            className="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
          >
            <div
              className="MuiList-root css-1mk9mw3-MuiList-root"
              id="country-detail--content--list-lang"
            >
              <li
                className="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1mhtu43-MuiListItem-root"
                disabled={false}
              >
                <div
                  className="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
                >
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="ArrowRightIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="m10 17 5-5-5-5v10z"
                    />
                  </svg>
                </div>
                <div
                  className="MuiListItemText-root css-tlelie-MuiListItemText-root"
                >
                  <span
                    className="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                  >
                    English
                  </span>
                </div>
              </li>
            </div>
          </div>
        </div>
      </div>
      <li
        className="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
        disabled={false}
      >
        <div
          className="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium css-ci5apu-MuiSvgIcon-root"
            data-testid="LocationCityIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15 11V5l-3-3-3 3v2H3v14h18V11h-6zm-8 8H5v-2h2v2zm0-4H5v-2h2v2zm0-4H5V9h2v2zm6 8h-2v-2h2v2zm0-4h-2v-2h2v2zm0-4h-2V9h2v2zm0-4h-2V5h2v2zm6 12h-2v-2h2v2zm0-4h-2v-2h2v2z"
            />
          </svg>
        </div>
        <div
          className="MuiListItemText-root css-tlelie-MuiListItemText-root"
        />
      </li>
    </ul>
  </div>
</div>
`;
